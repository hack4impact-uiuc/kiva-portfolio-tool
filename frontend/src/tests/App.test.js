import React from 'react'
import ReactDOM from 'react-dom'
import App from '../components/App'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('Basic whole app rendering with snapshot', () => {
  // create component to be tested
  const component = renderer.create(<App />)

  // change react component into pure javascript to compare to snapshot
  // checks if the component is the same as the snapshot (nothing unexpected happened)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
