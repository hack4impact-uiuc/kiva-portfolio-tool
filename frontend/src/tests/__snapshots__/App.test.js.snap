// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic whole app rendering with snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
